### ISSUE TITLE
Accuracy of integration as target

### ISSUE THREAD

**Issue by mayou36** [N3PDF/vegasflow#72]
### The problem

We would like to integrate a function for zfit that integrates a (blackbox) function up to a certain accuracy. So we are not interested in the number of dimensions but would like the algorithm to continue until the accuracy is reached (or maybe a `max_iter` value). Is that too much black-magic? Or what do you think?

### Proposed solution

So far I am not quite sure on the internals and maybe it's also worth to have a new high level function for this logic.

### Are you available/want to contribute?

Yes

**Comment by scarlehoff** [comment]
Let me see if I understood correctly:

Right now in order to integrate you have to provide a number of iterations and a number of events per iterations.

What you would like instead is to provide a target accuracy (say, 1%) and let the program iterate until said accuracy is reached, wright? (maybe providing a max. number of iterations so it doesn't hang forever).
And maybe, if after reaching the maximum number of iterations the desired accuracy is not obtained, bump the number of events per iterations and start again (also with some max.)

I think both would be doable as high level functions as you propose. Although maybe it also makes sense as part of the interface. I'll give it some thought, thanks for the suggestion. 

It shouldn't even require too much black magic. To first approximation you want an `if` here: https://github.com/N3PDF/vegasflow/blob/d79a18fc7e3e677d950483dca68c4b6281882364/src/vegasflow/monte_carlo.py#L496 and if whatever desired precision is not found just go back to the beginning*

**Comment by jonas-eschle** [comment]
Yes indeed, that would be approximately my thought, just to make sure that this also runs within the tensorflow graph if possible, something like inside a `while_loop`
